src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -0,0 +1,126 @@
+package com.google.sps.data;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/** File to run the GET request in java before putting it in the main adaptor function. 
+ */
+public class NSWGovAdaptor {
+
+  private HashMap<Integer, GovSiteDetails> dataMap;
+
+  public NSWGovAdaptor() throws Exception {
+    dataMap = new HashMap<>();
+    this.updateSiteInfo();
+  }
+
+  public HashMap<Integer, GovSiteDetails> getMap(HashMap<Integer, GovSiteDetails> dataMap) {
+    return this.dataMap;
+  }
+
+  public Coordinates getCoord(HashMap<Integer, GovSiteDetails> Map, Integer siteId) {
+    return new Coordinates(Map.get(siteId).lng, Map.get(siteId).lat);
+  }
```

Comment:
```suggestion
  public Coordinates getCoord(Integer siteId) {
    if map.containsKey(siteId) {
      return new Coordinates(this.dataMap[siteId].lng, this.dataMap[siteId].lat);
    }
    return null; // TODO(rosanna), take null return into consideration or raise exception
  }
```

Reply:
ah, i actually have written a since updated version of this using containsKey similar to your suggestion with raising exception and using containsKey,  but havent yet pushed.

Reply:
ah, i actually have written a since updated version of this using containsKey similar to your suggestion with raising exception and using containsKey, but haven't yet pushed. But will definitely take your suggestions on board and reflect it, thank you!